crates/router/src/services/authentication/decision.rs

```diff
@@ -0,0 +1,249 @@
+const RULE_DELETE_METHOD: common_utils::request::Method = common_utils::request::Method::Delete;
+
+pub const REVOKE: &str = "REVOKE";
+pub const ADD: &str = "ADD";
+
+// # Types
+//
+
+/// [`RuleRequest`] is a request body used to register a new authentication method in the proxy.
+#[derive(Debug, Serialize)]
+pub struct RuleRequest {
+    /// [`tag`] similar to a partition key, which can be used by the decision service to tag rules
+    /// by partitioning identifiers. (e.g. `tenant_id`)
+    pub tag: String,
+    /// [`variant`] is the type of authentication method to be registered.
+    #[serde(flatten)]
+    pub variant: AuthRuleType,
+    /// [`expiry`] is the time **in seconds** after which the rule should be removed
+    pub expiry: Option<u64>,
+}
+
+#[derive(Debug, Serialize)]
+pub struct RuleDeleteRequest {
+    pub tag: String,
+    #[serde(flatten)]
+    pub variant: AuthType,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum AuthType {
+    /// [`ApiKey`] is an authentication method that uses an API key. This is used with [`ApiKey`]
+    ApiKey { api_key: Secret<String> },
+}
+
+#[derive(Debug, Serialize)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum AuthRuleType {
+    /// [`ApiKey`] is an authentication method that uses an API key. This is used with [`ApiKey`]
+    /// and [`PublishableKey`] authentication methods.
+    ApiKey {
+        api_key: Secret<String>,
+        identifiers: Identifiers,
+    },
+}
+
+#[allow(clippy::enum_variant_names)]
+#[derive(Debug, Serialize, Clone)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum Identifiers {
+    /// [`ApiKey`] is an authentication method that uses an API key. This is used with [`ApiKey`]
+    ApiKey { merchant_id: String, key_id: String },
+    /// [`PublishableKey`] is an authentication method that uses a publishable key. This is used with [`PublishableKey`]
+    PublishableKey { merchant_id: String },
+}
+
+// # Decision Service
+//
+
+pub async fn add_api_key(
+    state: &SessionState,
+    api_key: Secret<String>,
+    merchant_id: String,
+    key_id: String,
+    expiry: Option<u64>,
+) -> CustomResult<(), ApiClientError> {
+    let decision_config = if let Some(config) = &state.conf.decision {
+        config
+    } else {
+        return Ok(());
+    };
+
+    let rule = RuleRequest {
+        tag: state.tenant.name.clone(),
+        expiry,
+        variant: AuthRuleType::ApiKey {
+            api_key,
+            identifiers: Identifiers::ApiKey {
+                merchant_id,
+                key_id,
+            },
+        },
+    };
+
+    let mut request = common_utils::request::Request::new(
+        RULE_ADD_METHOD,
+        &(decision_config.base_url.clone() + DECISION_ENDPOINT),
+    );
+
+    request.set_body(RequestContent::Json(Box::new(rule)));
+    request.add_default_headers();
+
+    let response = state
+        .api_client
+        .send_request(state, request, None, false)
```

Comment:
@akshay-97 , we cannot pass standalone `ApiClient` as the `send_request` function takes `SessionState` as an input argument